chapter/related-work.md

```diff
@@ -12,27 +12,102 @@ Additionally, this part aims to recognize alternative approaches to the LSP, in
+
+The work presented by Leimeister in [@cpachecker-lsp] exemplifies how functionality can be provided by entirely external tools as well.
+The server can be used to automatically perform software verification in the background using CPAchecker[@cpachecker].
+CPAchecker is a platform for automatic, extensible, software verification.
+The program is written in Java and provides a command line interface to be run locally.
+Additionally, it is possible to execute resource intensive verification through an HTTP-API on more powerful machines or clusters [@cpa-google-cloud,@cpa-clusters].
+The LSP-Server supports both modes of operation.
+On one hand it interfaces directly with the Java modules provided by the CPAchecker library, on the other it is able to utilize an HTTP-API provided by a server instance of the verifier.
+
+Similar to the work by Leimeister, in [@comprehension-features] Mészáros et al. present a proof of concept leveraging the LSP to integrate (stand-alone) code comprehension tools with the LSP compliant VSCode editor.
+Code comprehension tools support the work with complex code bases by "providing various textual information, visualization views and source code metrics on multiple abstraction levels".
+Pushing the boundaries of LSP use-cases, code comprehension tools do not only analyze specific source code, but also take into account contextual information.
+One of such tools is CodeCompass [@code-compass].
+The works of Mészáros yielded a language server that allowed to access the analysis features of CodeCompass in VSCode.
+In their paper they specifically describe the generation of source code diagrams.
+Commands issued by the client are processed by a CodeCompass plugin which acts as an LSP server and interacts with CodeCompass through internal APIs.
+
+
+#### Language Servers for Domain Specific Languages
+
+Bünder and Kuchen [@multi-editor-support] highlight the importance of the LSP in the area of Domain Specific Languages (DSL).
+Compared to general purpose languages, DSLs often focus both technical and non-technical users.
+While DSL creation workbenches like Xtext [@eclipse-xtext], Spoofax [@spoofax] or MPS[@jetpbrains-mps] allow for the implementation and provision of Eclipse or IntelliJ based DSLs, tooling for these languages is usually tied to the underlying platform.
+Requiring a specific development platform does not satisfy every user of the language.
+Developers may prefer using different editors, while non-technical users could easily be overwhelmed by a highly technical tool such as Eclipse and likewise prefer a lighter editor or integration into their business application.
+The authors of [@multi-editor-support] present how Xtext can be employed to forge DSLs that are relying on the LSP to provide multi-editor support.
+Using a generated LSP server allows the integration of DSL language tooling into LSP capable editors.
```

Comment:
It sounds redundant with the previous sentence.

Reply:
Yes agree
```suggestion

```